myproject/supermarket_manager/views.py

```diff
@@ -32,3 +32,60 @@ def add_role(request):
         return JsonResponse({"message": REQUIRED}, status=400)
 
     return JsonResponse({"message": INVALID_METHOD}, status=405)
+
+
+def list_roles(request):
+    """
+    API endpoint to retrieve a list of roles from the database.
+
+
+    This function handles GET requests to retrieve a list of roles from the database.
+
+
+    Returns:
+        JsonResponse: A JSON response containing the list of roles.
+    """
+
+    if request.method == 'GET':
+        roles = Role.objects.all()
```

Comment:
Would you consider pagination, please add the commented code. Since this code will take all the roles, it may not be good for performance problems.